tratto/src/main/java/star/tratto/util/javaparser/TypeUtils.java

```diff
@@ -0,0 +1,295 @@
+     * @return the ClassGetName form of the component type without arrays
+     */
+    private static String classGetNameComponentType(String classGetNameArray) {
+        int arrayLevel = getArrayLevel(classGetNameArray);
+        if (arrayLevel == 0) {
+            return classGetNameArray;
+        }
+        String classGetNameComponent = classGetNameArray.replaceAll("\\[", "");
+        if (isPrimitiveFieldDescriptor(classGetNameComponent)) {
+            return classGetNameComponent;
+        } else {
+            classGetNameComponent = classGetNameComponent.replaceAll(";", "");
+            // for arrays of Objects, remove "L" prefix
+            return classGetNameComponent.substring(1);
+        }
+    }
+
+    /**
+     * Converts a ClassGetName form of a type to its corresponding
+     * ClassGetSimpleName form.
+     *
+     * @param classGetName a ClassGetName form of a type
+     * @return the corresponding ClassGetSimpleName form of a type
+     */
+    private static String classGetNameToClassGetSimpleName(
+            String classGetName
+    ) {
+        int arrayLevel = getArrayLevel(classGetName);
+        classGetName = classGetNameComponentType(classGetName);
+        if (isPrimitiveFieldDescriptor(classGetName)) {
+            // ClassGetName uses field descriptors for arrays of primitive types
+            classGetName = Signatures.fieldDescriptorToBinaryName(classGetName);
+        } else {
+            classGetName = getInnermostClassNameFromClassGetName(classGetName);
+        }
+        return addArrayLevel(classGetName, arrayLevel);
+    }
+
+    /**
+     * Converts a list of ClassGetName forms of types to their corresponding
+     * ClassGetSimpleName forms.
+     *
+     * @param classGetNames ClassGetName forms of types
+     * @return the corresponding ClassGetSimpleName forms of types
+     */
+    public static List<String> classGetNameToClassGetSimpleName(
+            List<String> classGetNames
+    ) {
+        return classGetNames
+                .stream()
+                .map(TypeUtils::classGetNameToClassGetSimpleName)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Removes type arguments and semicolons from a parameterized type name.
+     * The given type name should originate from source code, as neither
+     * ClassGetSimpleName nor ClassGetName forms include type parameters.
+     * These names may include semicolons due to source code format. For
+     * example,
+     *     "private final static Map&lt;String, Integer&gt;;"
+     * includes a semicolon at the end of its declaration, which is removed
+     * by this method.
+     *
+     * @param parameterizedType a type name from source code
+     * @return the base type without type arguments
+     */
+    public static String removeTypeArgumentsAndSemicolon(String parameterizedType) {
+        String regex = "<[^<>]*>|;";
+        // repeatedly remove all type arguments.
+        String previous;
```

Comment:
Declare variables in the smallest scope possible.  For example, declare this within the loop to signal to the reader that it does not contribute to any loop-carried dependencies.